### PR TITLE
Docker build failure - Add Tini and set version range for numpy

### PR DESCRIPTION
**Pull request by fzagarzazu**
First time user, I spent some time setting it up and I ran into a few issues, the purpose of this pull request is to bring it to your attention, maybe it will help somebody else. Please close it if it doesn't apply.

### Background:
- First time user.
- OS X El Capitan + VirtualBox
- Docker version 1.9.1,

### Docker build failure:

```
Obtaining file:///zipline
    Complete output from command python setup.py egg_info:
    warning: no files found matching '*.pyx' under directory 'Cython/Debugger/Tests'
    warning: no files found matching '*.pxd' under directory 'Cython/Debugger/Tests'
    warning: no files found matching '*.h' under directory 'Cython/Debugger/Tests'
    warning: no files found matching '*.pxd' under directory 'Cython/Utility'
    Unable to find pgen, not compiling formal grammar.
    
    Installed /zipline/.eggs/Cython-0.24.1-py2.7-linux-x86_64.egg
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/zipline/setup.py", line 300, in <module>
        **conditional_arguments
      File "/usr/local/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/local/lib/python2.7/site-packages/setuptools/dist.py", line 315, in __init__
        self.fetch_build_eggs(attrs['setup_requires'])
      File "/usr/local/lib/python2.7/site-packages/setuptools/dist.py", line 361, in fetch_build_eggs
        replace_conflicting=True,
      File "/usr/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 851, in resolve
        dist = best[req.key] = env.best_match(req, ws, installer)
      File "/usr/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1116, in best_match
        dist = working_set.find(req)
      File "/usr/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 720, in find
        raise VersionConflict(dist, req)
    pkg_resources.VersionConflict: (numpy 1.9.2 (/usr/local/lib/python2.7/site-packages), Requirement.parse('numpy>=1.11.1'))
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /zipline/
```

### Possible solution / workaround:

- Set numpy version range. _(included in commit)._

### Jupyter notebooks: Dead kernels / automatic restart failures.

- Please see https://github.com/ipython/ipython/issues/7062 for details.
- Possible solution: Add Tini - https://github.com/krallin/tini. _(included in commit)._

Thank you for your work.

### CODE REVIEW

**Comment by lssilva** [comment]
Thank you I had the same issue, I hope this is integrated soon

File: Dockerfile [hunk 1]
```diff
@@ -49,13 +54,14 @@ RUN mkdir ${PROJECT_DIR} \
 
 WORKDIR /ta-lib
 
-RUN pip install numpy==1.9.2 \
+RUN pip install 'numpy>=1.11.1,<2.0.0' \
```
Comment (ssanderson) [inline Dockerfile]: Latest zipline is also on pandas 0.18.1 and scipy 0.17.1 if you want to bump those as well.

File: Dockerfile [hunk 2]
```diff
@@ -26,6 +26,11 @@ FROM python:2.7
 #
 # set up environment
 #
+ENV TINI_VERSION v0.10.0
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
+RUN chmod +x /tini
+ENTRYPOINT ["/tini", "--"]
```
Comment (ssanderson) [inline Dockerfile]: Is there a good executive summary somewhere of the benefits of using `tini` as the container pid 1?  I'm aware vaguely that it solves issues with zombie processes not being correctly reaped, but I don't have a clear sense of concrete cases where it fixes issues.

File: Dockerfile (hunk 1 above)
Comment (fzagarzazu) [inline Dockerfile]: Good point, latest commit sets the version range for pandas and scipy.

File: Dockerfile (hunk 2 above)
Comment (fzagarzazu) [inline Dockerfile]: I didn't dig much into it, but this is the best explanation I found https://github.com/krallin/tini/issues/8 .

File: Dockerfile (hunk 1 above)
Comment (fzagarzazu) [inline Dockerfile]: Please note that I added `  && pip install matplotlib \` to the Dockerfile in order to fix an error when executing _zipline/examples/buyapple.ipynb_ .

File: Dockerfile (hunk 2 above)
Comment (fzagarzazu) [inline Dockerfile]: For what it's worth I am running it with _docker-compose_ .

**Comment by ssanderson** [comment]
:+1:, thanks!

**Pull request merged**